### DISCUSSION TITLE
Notion 如果一篇文章的内容改动想重新发布有什么自动化的方案吗，根据标签，设置为“改动后重发布” 可行嘛

### DISCUSSION THREAD

**Discussion by Timeyangzai** [LetTTGACO/elog#109] [Q&A] [answered]
```
module.exports = {
  write: {
    platform: 'notion',
    notion: {
      token: process.env.NOTION_TOKEN,
      databaseId: process.env.NOTION_DATABASE_ID,
      filter: {property: 'status', select: {equals: '已发布'}},
```

导出模块内我是这样写的，发布的内容会进入到elog.cache.json缓存里面，有没有什么办法让改动的内容重新再获取发布一次呀

**Comment by LetTTGACO** [comment]
自动化配置可以参考这里：https://elog.1874.cool/notion/vy55q9xwlqlsfrvk

**Reply by LetTTGACO** [reply]
elog 本身就是增量更新，所以当 Notion 文章改动，重新运行 elog sync 就会重新同步改动的文章。自动化的过程就是自动运行elog sync，可以通过 Github Action 的 API 来触发此命令，所以需要监听 Notion 的字段变动（Notion 目前只支持 Slack，所以需要通过 Slack + pipedream 配置来调用 Github Action 的 API）

**Reply by Timeyangzai** [reply]
好的 晚点我试一下Slack的操作，还有一个问题，我想自定义一下NotionToMarkdown模块中的setCustomTransformer，让hexo网站适配bookmark的格式返回html格式去替换，而不是[bookmark]（网址）；这个格式，有没有什么接口去调用啊，还是说只能用hexo部署中的`hexo.extend.filter.register("after_post_render"）`去后处理呀

**Reply by LetTTGACO** [reply, marked as answer]
可以在拓展点中替换，用正则表达式匹配[bookmark]（网址）这个格式，然后重新格式化为你想要的。
Elog 提供[拓展点](https://elog.1874.cool/notion/deploy-platform#formatext-%E5%AD%97%E6%AE%B5%E8%AF%B4%E6%98%8E)，你可以拿到正文内容自行替换

大概思路
<img width="779" alt="image" src="https://github.com/user-attachments/assets/87f41fe2-ad12-4b7c-ac2b-1d38495aea1f">

**Reply by Timeyangzai** [reply]
吼吼，可以了，就是markdown中还要写{% raw %} {% endraw %}，研究了好久...，要是notion字体、字体背景、最新出的mention啥的能导出，让我自己正则就好了 哈哈，感觉能用网页还原notion了
![image](https://github.com/user-attachments/assets/fa953860-b974-4e40-89cf-6ea0dbb01e8e)

**Reply by LetTTGACO** [reply]
如果要最大程度还原 Notion 可以试试 隔壁的 NotionNext。再配合 Elog 用来备份也不是不行，我之前就这么搞的。后来觉得 markdown 就足够，内容最重要，所以就换回来了，然后用 Astro 渲染
